app/src/main/java/org/simple/clinic/summary/addphone/AddPhoneNumberUiRender.kt

```diff
@@ -1,9 +1,22 @@
 package org.simple.clinic.summary.addphone
 
 import org.simple.clinic.mobius.ViewRenderer
+import org.simple.clinic.registration.phone.PhoneNumberValidator.Result.Blank
+import org.simple.clinic.registration.phone.PhoneNumberValidator.Result.LengthTooLong
+import org.simple.clinic.registration.phone.PhoneNumberValidator.Result.LengthTooShort
+import org.simple.clinic.registration.phone.PhoneNumberValidator.Result.ValidNumber
 
 class AddPhoneNumberUiRender(private val ui: AddPhoneNumberUi) : ViewRenderer<AddPhoneNumberModel> {
   override fun render(model: AddPhoneNumberModel) {
+    renderValidationErrors(model)
+  }
 
+  private fun renderValidationErrors(model: AddPhoneNumberModel) {
+    when (val result = model.validationResult) {
+      is LengthTooLong -> ui.showPhoneNumberTooLongError(result.maximumRequiredNumberLength)
```

Comment:
`AddPhoneNumberModel#validationResult` can be `null`, right? This case is not being handled here.

Reply:
Ah right, Good catch 👍